src/cryptonote_core/service_node_rules.cpp

```diff
@@ -280,4 +280,18 @@ bool get_portions_from_percent_str(std::string cut_str, uint64_t& portions) {
   return false;
 }
 
+uint64_t get_atomic_amount_from_percent_str(std::string cut_str) {
+  const auto pct = parse_fee_percent(cut_str);
+  if (!pct)
+    throw "could not parse fee percent";
+
+  if(*pct < 0.0 || *pct > 100.0)
+    throw "fee percent out of bounds";
```

Comment:
This needs an exception type (I'm sad that being able to throw a raw c string pointer even compiles in C++).

Reply:
Probably `std::out_of_range` for this one, and `std::invalid_argument` for the first one.